api-http/proposals/snapshot-api.md

```diff
@@ -0,0 +1,566 @@
+
+
+| **API URL** 	| **Purpose** 	| **Parameters** 	|
+|----------	|----------	|----------	|
+| /producer/schedule_snapshot         	| Adds a task to perform snapshot         	| start_block, end_block, blocks_count, snapshot_description     	|
+| /producer/get_snapshot_requests         	| Returns scheduled snapshot requests         	| start_snapshot_request_id, limit (number of snapshots to return)         	|
+| /producer/get_snapshot_request_status         	| Queries status of a snapshot request        	| Snapshot ID         	|
+| /producer/unschedule_snapshot         	| Removes previously scheduled snapshot task         	| Snapshot ID         	|
+
+### Scheduling Snapshot
+- request takes 4 optional parameters - start_block, end_block, blocks_count, snapshot_description:
+    - start_block: first block at which snapshot will be taken, if not set, will be taken immediately similar to an old create_snapshot call
+    - end_block: block number at which schedule for the snapshot will be no longer active and snapshot task will be automatically removed. If end_block is not specified, snapshot task will never be auto removed and needs to be unscheduled with another api call to stop
+    - blocks_count - number of blocks after which a recurring snapshot will be made. If not set, snapshot will be done only once and scheduled task will be removed
+    - snapshot_description - optional description of the snapshot
+- in case of success api call should return json describing scheduled snapshot task which should contain unique snapshot request id
+- in case of error a descriptive message and error code should be returned
+
+As an additional consideration when scheduling a snapshot should be it's alignment with previously scheduled snapshots. Some examples of possible cases:
+- Duplicate snapshot requests should be rejected
+- Requests that has different parameters but will result in executing snapshots at the same block num as previously scheduled snapshots should result in a single snapshot for block id
+
+Please note that single snapshot request can trigger and correspond to multiple pending snapshots due to waiting for irreversibility or in the case where snapshots happenning on different forks
+
+:bulb: Implementation note - It would be great to encapsulate this logic into a separate class, so we can tweak/alter it without touching other code
+
+![image](https://user-images.githubusercontent.com/79997103/212993658-af39ac43-ee64-4578-8d8f-4cbe16dba1da.png)
+
+### Retrieving Snapshot Requests
+This endpoint will return all scheduled snapshot requests with pagination - starting from snapshot determined by 'start_snapshot_request_id' (optional) and returning 'limit' number of snapshots
+
+![image](https://user-images.githubusercontent.com/79997103/212993725-debd027e-05d0-490b-bbe4-ee95e5d1820e.png)
+
+### Retrieving Snapshot Status
+Request takes snapshot_id as parameter and either returns object describing scheduled snapshot or an error
```

Comment:
Can status show which ones are in progress, which ones are pending? For in progress, can the number of blocks being done shown?

Reply:
A pending snapshot is one in progress. Do you see a distinction between the two?

Reply:
OK, I see.